src/Cli.Tests/UtilsTests.cs

```diff
@@ -234,6 +234,64 @@ public void TestMergeConfig()
             }
         }
 
+        /// <summary>
+        /// Test to verify that merged config file is only used for the below scenario
+        /// 1. Environment value is set.
+        /// 2. Both Base and envBased config file is present.
+        /// In all other cases, the TryMergeConfigsIfAvailable method should return false
+        /// and out string for the mergedConfigFile should be empty.
+        /// </summary>
+        [DataTestMethod]
+        [DataRow("", false, false, "", false, DisplayName = "If environment value is not set, merged config file is not generated.")]
+        [DataRow("", false, true, "", false, DisplayName = "If environment value is not set, merged config file is not generated.")]
+        [DataRow("", true, false, "", false, DisplayName = "If environment value is not set, merged config file is not generated.")]
+        [DataRow("", true, true, "", false, DisplayName = "If environment value is not set, merged config file is not generated.")]
+        [DataRow("Test", false, false, "", false, DisplayName = "Environment value set but base config not available, merged config file is not generated.")]
+        [DataRow("Test", false, true, "", false, DisplayName = "Environment value set but base config not available, merged config file is not generated.")]
+        [DataRow("Test", true, false, "", false, DisplayName = "Environment value set but env based config not available, merged config file is not generated.")]
+        [DataRow("Test", true, true, "dab-config.Test.merged.json", true, DisplayName = "Environment value set and both base and envConfig available, merged config file is generated.")]
+        public void TestMergeConfigAvailability(
+            string environmentValue,
+            bool isBaseConfigPresent,
+            bool isEnvironmentBasedConfigPresent,
+            string expectedMergedConfigFileName,
+            bool expectedIsMergedConfigAvailable)
+        {
+            Environment.SetEnvironmentVariable(RUNTIME_ENVIRONMENT_VAR_NAME, environmentValue);
```

Comment:
Does the CLI act differently if the environment variable KEY is not present? this function call sets the key but may leave the value empty, which is slightly different than the key not existing.

Reply:
Thanks, Added tests for the same.